Use AWS CRT instead of cryptography for Cloudfront url signing

kdaily opened the pull request:
*Issue #, if available:*

- #9258
- #9109 

*Description of changes:*

Fixes issue from #9258 to support PKCS8-formatted private key PEM files. The support for this was added directly to `awscrt==0.25.6` (https://github.com/awslabs/aws-crt-python/pull/638/). This PR bumps CRT to 0.26.1.

Added a test to prevent regression for support of this format. I also manually ran `aws cloudfront sign` commands and confirmed that the expected signature is generated using both a PKCS1 and PKCS8 pem file.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

kdaily commented on requirements/download-deps/portable-exe-lock.txt:
```diff
@@ -202,9 +202,9 @@ pyinstaller==6.11.1 \
-pyinstaller-hooks-contrib==2025.1 \
-    --hash=sha256:130818f9e9a0a7f2261f1fd66054966a3a50c99d000981c5d1db11d3ad0c6ab2 \
-    --hash=sha256:d3c799470cbc0bda60dcc8e6b4ab976777532b77621337f2037f558905e3a8e9
+pyinstaller-hooks-contrib==2025.2 \
```
Meant to roll back this separate change that came in when regenerating the lockfiles.

kdaily replied:
Fixed in f2a02f82568cfcbf1ae8514f5ef6c992ff9b9834

ashovlin commented on pyproject.toml:
```diff
@@ -41,7 +41,7 @@ dependencies = [
     "prompt-toolkit>=3.0.24,<3.0.39",
     "distro>=1.5.0,<1.9.0",
-    "awscrt==0.25.4",
+    "awscrt==0.25.6",
```
From that internal chat, should we take this to 0.25.7 now for the improved debugging?

kdaily replied:
Correct, I'm bumping that version and testing with an internal release.

ashovlin commented on .changes/next-release/enhancement-cloudfront-1337.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "type": "enhancement",
+  "category": "``cloudfront``",
+  "description": "Replace cryptographic functions from ``cryptography`` with ``awscrt`` for the ``sign`` command. Fixes https"
```
Looks like a fragment for the link?

kdaily replied:
Yikes, not sure what happened there. Thanks!

ashovlin commented:
Does this still need the [changes to cloudfront.py](https://github.com/aws/aws-cli/pull/9109/files#diff-f24f8c7ad456198aafe4f85cd73918c941bb38ff33a7ffc70bf856af98ed05b9) that were originally in #9109, since we reverted to all of the `cryptography.hazmat`?

kdaily commented:
> Does this still need the [changes to cloudfront.py](https://github.com/aws/aws-cli/pull/9109/files#diff-f24f8c7ad456198aafe4f85cd73918c941bb38ff33a7ffc70bf856af98ed05b9) that were originally in #9109, since we reverted to all of the `cryptography.hazmat`?

Yeah. Not sure how I missed pushing that commit.

kdaily commented:
> > Does this still need the [changes to cloudfront.py](https://github.com/aws/aws-cli/pull/9109/files#diff-f24f8c7ad456198aafe4f85cd73918c941bb38ff33a7ffc70bf856af98ed05b9) that were originally in #9109, since we reverted to all of the `cryptography.hazmat`?
> 
> Yeah. Not sure how I missed pushing that commit.

Fixed in a5b685b91fbc4ba942882fa52f97a9d6c6f92b84
